### ISSUE TITLE
Server memory usage displayed on Panel not matching what docker reports

### ISSUE THREAD

**Issue by jdlovins** [Pterodactyl/Panel#658]
I noticed that the server memory use displayed on the panel doesn't quite match what docker actually reports.

It seems like the lib you are using dockerode uses docker stats but with the API call instead as found here; https://github.com/apocas/dockerode/blob/d2fdfa5712e4f6e63e8131c3fc2b33275ab20b6a/lib/container.js#L1022

However here is the output of using ``docker stats <containerId>``
```
CONTAINER           CPU %               MEM USAGE / LIMIT    MEM %               NET I/O             BLOCK I/O           PIDS
58d                 28.12%              650.6MiB / 1023MiB   63.57%              783MB / 1.38GB      17.6GB / 46.5MB     26
```

and here is what the panel reports, 
![screen shot 2017-10-04 at 12 05 10 pm](https://user-images.githubusercontent.com/14537284/31186724-77c6e614-a8fd-11e7-8b99-704bc76aab53.png)

CPU usage seems to be on point though.

**Comment by DaneEveritt** [comment]
Thats... interesting. I've always had it be pretty much dead on. 😕

**Comment by jdlovins** [comment]
Anything i can try to do to debug it?

**Comment by DaneEveritt** [comment]
I'm not sure. It sounds like this is just one server that is reporting wrongly?

**Comment by jdlovins** [comment]
I think it's just tf2 things. It reports incorrectly with all tf2 servers but correctly with css i believe. ``docker stats <containerid>`` is accurate though, but that would be a pain to implement. I see no easy fix so do what you will with this.

**Comment by DaneEveritt** [comment]
I already report from docker stats.

**Comment by jdlovins** [comment]
Not from what I saw, unless I missed something you use dockerode which uses the docker engine API. That returns different results than ``docker stats <ctid>``

**Comment by DaneEveritt** [comment]
The daemon uses Dockerode, yes. But Dockerode is simply a wrapper for dockers official API.

https://github.com/Pterodactyl/Daemon/blob/83f3cded7bd984e645444007badd317e9741c710/src/controllers/docker.js#L253-L267

That code above calls Dockerode's `stats` function, which then wraps around Docker's official API and streams the stats continuously. There should be no difference in the data being returned, and I've never seen it return differently. This is especially odd considering it is only broken for a few specific servers. I'm not really sure what to tell you, but you're the first person to encounter this issue across hundreds of other deployments and probably thousands of running containers reporting stats.

**Comment by jdlovins** [comment]
Right, I actually downloaded dockerode and played around with it to see if I could come across any reason why it would be showing different stats. It returned what the panel reported of course, but cli docker stats was returning different (but matching what top showed). So I'm actually quite curious why the docker engine API returns x and ``docker stats`` returns y. 
I do believe ``docker stats`` is more accurate overall for some reason but like I said it's probably not worth implementing directly. 

As for why im the outlier in this all, I have no idea. I simply installed everything per docs on centOS and ubuntu and it's returning like that. I guess i could try to reach out to some sort of docker discord / irc channel and see if i can get any clues from there.
